File: Data Structures & Algorithms/products-of-array-discluding-self/submission-2.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int n = nums.length;
        int[] sol = new int[n];
        sol[0] = 1;
        for (int i = 1; i < n; i++) {
            sol[i] = sol[i-1]*nums[i-1];
        }
        int temp = 1;
        for (int i = n-1; i >= 0; i--) {
            sol[i] = sol[i]*temp;
            temp*=nums[i];
        }

        return sol;
    }
}  

File: Data Structures & Algorithms/remove-element/submission-2.go
func removeElement(nums []int, val int) int {
    n := len(nums)

	for i := 0; i < n;  {
		if nums[i] == val {
			n--
			nums[i] = nums[n]
		} else {
			i++
		}
	}
	return n
}
